cclib/parser/molcasparser.py

```diff
@@ -686,6 +686,74 @@ def extract(self, inputfile, line):
                         'symmetry. Ignoring these coordinates.'
                         % (len(atomcoords), self.natom))
 
+        ## Parsing Molecular Gradients attributes in this section.
+        # ()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()
+        # 
+        #                                               &ALASKA
+        # 
+        #                                    only a single process is used
+        #                        available to each process: 2.0 GB of memory, 1 thread
+        # ()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()()
+        # ...
+        # ...
+        #  **************************************************
+        #  *                                                *
+        #  *              Molecular gradients               *
+        #  *                                                *
+        #  **************************************************
+        # 
+        #   Irreducible representation: a  
+        #  ---------------------------------------------------------
+        #                      X             Y             Z        
+        #  ---------------------------------------------------------
+        #   C1               -0.00009983   -0.00003043    0.00001004
+        #   ...
+        #   H20              -0.00027629    0.00010546    0.00003317
+        #  ---------------------------------------------------
+        # WARNING: "Molecular gradients, after ESPF" is found for ESPF QM/MM calculations
+        if "Molecular gradients " in line:
+
+            if not hasattr(self, "grads"):
+                self.grads = []
+
+            self.skip_lines(inputfile, ['stars', 'stars', 'blank', 'header',
+                                        'dashes', 'header', 'dashes'])
+
+            grads = []
+            line = next(inputfile)
+            while len(line.split()) == 4:
+                tmpgrads = list(map(float, line.split()[1:]))
+                grads.append(tmpgrads)
+                line = next(inputfile)
+
+            self.grads.append(grads)
+
+        # This code here works, but QM/MM gradients are printed after QM ones.
+        # Maybe another attribute is needed to store them to have both.
+        if "Molecular gradients, after ESPF" in line:
+
+            if not hasattr(self, "_grads"):
+                self._grads = []
+
+            self.skip_lines(inputfile, ['stars', 'stars', 'blank', 'header',
+                                        'dashes', 'header', 'dashes'])
+
+            _grads = []
+            line = next(inputfile)
+            while len(line.split()) == 4:
+                tmpgrads = list(map(float, line.split()[1:]))
+                _grads.append(tmpgrads)
+                line = next(inputfile)
+
+            self._grads.append(_grads)
+
+        # Replace the main attribute with QM/MM gradients
```

Comment:
Could you just accumulate into a local variable in the block above, and then do `self.set_attribute('grads', grads)`? That is enough to ensure that the ESPF gradients are parsed and provided when available.

Reply:
I had to do a bit differently to overwrite all previous gradients parsed to be overwritten.